test/pytest/test_handler.py

```diff
@@ -182,124 +216,215 @@ def test_kserve_mnist_model_register_and_inference_on_valid_model():
+        "labels": info["action_labels"],
+    }
+    response = run_inference_using_url_with_data(
+        TF_INFERENCE_API + "/v1/models/MMF_activity_recognition_v2:predict",
+        pfiles=files,
+    )
     response = response.content.decode("utf-8")
     response = ast.literal_eval(response)
     response = [n.strip() for n in response]
-    assert response == ['Sitting at a table','Someone is sneezing','Watching a laptop or something on a laptop']
+    assert response == [
+        "Sitting at a table",
+        "Someone is sneezing",
+        "Watching a laptop or something on a laptop",
+    ]
     test_utils.unregister_model("MMF_activity_recognition_v2")
 
+
 def test_huggingface_bert_model_parallel_inference():
     number_of_gpus = torch.cuda.device_count()
     check = os.popen(f"curl http://localhost:8081/models")
     print(check)
     if number_of_gpus > 1:
         batch_size = 1
-        batch_delay = 5000 # 10 seconds
+        batch_delay = 5000  # 10 seconds
         params = (
-            ('model_name', 'Textgeneration'),
-            ('url', 'https://bert-mar-file.s3.us-west-2.amazonaws.com/Textgeneration.mar'),
-            ('initial_workers', '1'),
-            ('batch_size', str(batch_size)),
-            ('max_batch_delay', str(batch_delay))
+            ("model_name", "Textgeneration"),
+            (
+                "url",
+                "https://bert-mar-file.s3.us-west-2.amazonaws.com/Textgeneration.mar",
+            ),
+            ("initial_workers", "1"),
+            ("batch_size", str(batch_size)),
+            ("max_batch_delay", str(batch_delay)),
         )
         test_utils.start_torchserve(no_config_snapshots=True)
         test_utils.register_model_with_params(params)
-        input_text = os.path.join(REPO_ROOT, 'examples', 'Huggingface_Transformers', 'Text_gen_artifacts', 'sample_text_captum_input.txt')
-        
-        response = os.popen(f"curl http://127.0.0.1:8080/predictions/Textgeneration -T {input_text}")
+        input_text = os.path.join(
+            REPO_ROOT,
+            "examples",
+            "Huggingface_Transformers",
+            "Text_gen_artifacts",
+            "sample_text_captum_input.txt",
+        )
+
+        response = os.popen(
+            f"curl http://127.0.0.1:8080/predictions/Textgeneration -T {input_text}"
+        )
         response = response.read()
-        
-        assert  'Bloomberg has decided to publish a new report on the global economy' in response
-        test_utils.unregister_model('Textgeneration')
+
+        assert (
+            "Bloomberg has decided to publish a new report on the global economy"
+            in response
+        )
+        test_utils.unregister_model("Textgeneration")
+    else:
+        logger.info(
+            "Running model parallel inference requuires more than one gpu, number of available gpus on thi machine is: ",
+            number_of_gpus,
+        )
+
+
+def test_huggingface_opt_distributed_inference_deepspeed():
```

Comment:
Made similar feedback to @HamidShojanazeri but the way I'd like to see this tested is
1. Have a seperate file for distributed tests
2. Make sure tests are skipped unless a CUDA multi GPU machine is found
3. Create a new requirements.txt with optional dependencies including all the distributed libraries to test